test/functional/mempool_ephemeral_dust.py

```diff
@@ -0,0 +1,493 @@
+
+        dusty_tx = self.wallet.create_self_transfer_multi(fee_per_output=0, version=3)
+        self.add_output_to_create_multi_result(dusty_tx)
+
+        sweep_tx = self.wallet.create_self_transfer_multi(utxos_to_spend=dusty_tx["new_utxos"], version=3)
+
+        # Test doesn't work because lack of package feerates
+        test_res = self.nodes[0].testmempoolaccept([dusty_tx["hex"], sweep_tx["hex"]])
+        assert not test_res[0]["allowed"]
+        assert_equal(test_res[0]["reject-reason"], "min relay fee not met")
+
+        # And doesn't work on its own
+        assert_raises_rpc_error(-26, "min relay fee not met", self.nodes[0].sendrawtransaction, dusty_tx["hex"])
+
+        # But if we add modified fees, it would succeed
+        self.nodes[0].prioritisetransaction(txid=dusty_tx["txid"], dummy=0, fee_delta=COIN)
+        test_res = self.nodes[0].testmempoolaccept([dusty_tx["hex"]])
+        assert test_res[0]["allowed"]
+        self.nodes[0].prioritisetransaction(txid=dusty_tx["txid"], dummy=0, fee_delta=-COIN)
+
+        # And package evaluation succeeds
+        res = self.nodes[0].submitpackage([dusty_tx["hex"], sweep_tx["hex"]])
+        assert_equal(res["package_msg"], "success")
+        assert_mempool_contents(self, self.nodes[0], expected=[dusty_tx["tx"], sweep_tx["tx"]])
+
+        self.generate(self.nodes[0], 1)
+        assert_equal(self.nodes[0].getrawmempool(), [])
+
+    def test_node_restart(self):
+        self.log.info("Test that an ephemeral package is rejected on restart due to individual evaluation")
+
+        assert_equal(self.nodes[0].getrawmempool(), [])
+
+        dusty_tx = self.wallet.create_self_transfer_multi(fee_per_output=0, version=3)
+        self.add_output_to_create_multi_result(dusty_tx)
+
+        sweep_tx = self.wallet.create_self_transfer_multi(utxos_to_spend=dusty_tx["new_utxos"], version=3)
+
+        res = self.nodes[0].submitpackage([dusty_tx["hex"], sweep_tx["hex"]])
+        assert_equal(res["package_msg"], "success")
+        assert_equal(len(self.nodes[0].getrawmempool()), 2)
+        assert_mempool_contents(self, self.nodes[0], expected=[dusty_tx["tx"], sweep_tx["tx"]])
+
+        # Node restart; doesn't allow allow ephemeral transaction back in due to individual submission
+        # resulting in 0-fee
+        self.restart_node(0)
+        self.restart_node(1)
+        self.connect_nodes(0, 1)
+        assert_mempool_contents(self, self.nodes[0], expected=[])
+
+    def test_fee_having_parent(self):
+        self.log.info("Test that a transaction with ephemeral dust may not have base fee, and ignores modified")
+
+        assert_equal(self.nodes[0].getrawmempool(), [])
+
+        sats_fee = 1
+        dusty_tx = self.wallet.create_self_transfer_multi(fee_per_output=sats_fee, version=3)
+        self.add_output_to_create_multi_result(dusty_tx)
+        assert_equal(dusty_tx["tx"].vout[0].nValue, 5000000000 - sats_fee) # main output is not dust
```

Comment:
A little bit brittle, as it assumes the utxo selected is from a coinbase.
```suggestion
        assert_equal(int(COIN * dusty_tx["fee"]), sats_fee) # main output is not dust
```

Reply:
added suggestion and a direct assertion that the main output amount is >330 the taproot dust amount